### PR TITLE
Rip out create_note in favour of reading from .cookiecutter.json pattern

### PR DESCRIPTION
**Pull request by alexlatchford**
Done:

- [ZOMLINFRA-400](https://zbrt.atl.zillow.net/browse/ZOMLINFRA-400): Rip out `git note` implementation of template context persistence/recovery. Replace it with a local `.cookiecutter.json` implementation, see [this PR](https://gitlab.zgtools.net/zillow-offers/zoml/ml-infrastructure/archetype.py-ml/merge_requests/17) for the write side.

Rationale for this approach:

With the previous implementation it relied on an [upstream patch](https://github.com/cookiecutter/cookiecutter/pull/908) to `cookiecutter` itself to have it return the fully rendered `context`. That PR has been open for ~2.5 years without being merged so I decided against utilizing this approach which is fundamental to the [existing `create_note` approach](https://github.com/rmedaer/milhoja/blob/master/milhoja/milhoja.py#L131) used by `milhoja.`

### CODE REVIEW

File: milhoja/core.py [hunk 1]
```diff
@@ -71,97 +67,64 @@ def __enter__(self):
...
-            #        -> Let user resolve ?
-            #        -> Raise an error after analysis ?
-            assert self.repo.index.conflicts is None
+            # If there is a conflict we should error and let the user manually resolve it.
+            if self.repo.index.conflicts is not None:
+                raise MergeConflictException(
+                    f'Cannot merge the template commit ({branch.target}) with the current HEAD '
+                    f'({self.repo.head}). Please resolve them manually and run \'git commit\' '
+                    'to merge'
+                )
```
Comment (tmckay) [inline milhoja/core.py]: Does it make sense to raise an exception here? For 99% of the cases, we would expect a merge conflict. That doesn't seem like an exceptional condition to me. Maybe this is handled elsewhere though.

File: README.md [hunk 2]
```diff
@@ -7,6 +7,16 @@
 Milhoja is a tool in top of [Cookiecutter](https://github.com/audreyr/cookiecutter) which maintains
 directory templating with Git. The first goal of Milhoja is to provide *upgrade* feature to [Cookiecutter](https://github.com/audreyr/cookiecutter).
 
+## Prerequistes
+
+It is assumed that your cookiecutter template contains a `.cookiecutter.json` file at the root directory, or you can override it's location by
```
Comment (shoelsch) [inline README.md]: Nit: `override it's location` -> `override its location`

File: README.md [hunk 3]
```diff
@@ -7,6 +7,16 @@
...
+passing in `--context-file`. Please use the [`jsonify`](https://github.com/cookiecutter/cookiecutter/pull/791) Jinja2 extension to dump the
+`cookiecutter` template context to `.cookiecutter.json`.
+
+**Tip:** One problem `milhoja` has that as divergence between the cookiecutter template and the project itself increase as will the volume of
+conflicts needed to be manually resolved for each upgrade merge. To minimize these it is often advisable to fit templates with a
+`generate_example` boolean flag which will disable including any example code, instead replacing implementation with a [`pass`](https://docs.python.org/2.0/ref/pass.html) for example.
```
Comment (shoelsch) [inline README.md]: Python 3 link?

File: milhoja/core.py (hunk 1 above)
Comment (alexlatchford) [reply milhoja/core.py]: I'm raising an exception in lieu of anything better to do at this point as we cannot automatically proceed unfortunately. Maybe we can handle and display this message a little nicer at the `cli.py` layer but to me it still seems like an exception even if it's going to be the default behavior essentially.

Comment (alexlatchford) [reply milhoja/core.py]: Cool after a discussion offline I decided to create https://zbrt.atl.zillow.net/browse/ZOMLINFRA-427 to handle this exception at the CLI layer more nicely and give the user instructions on the best path forward in this case.

**Pull request merged**